tribestream-api-registry-webapp/src/main/java/org/tomitribe/tribestream/registryng/resources/ApplicationResource.java

```diff
@@ -89,11 +90,17 @@ public Response createService(
             ApplicationWrapper application) {
         final Swagger swagger = application.getSwagger();
         validate(swagger);
-        final OpenApiDocument document = repository.insert(swagger);
-        final OpenApiDocument newDocument = repository.findByApplicationIdWithEndpoints(document.getId());
-        return Response.status(Response.Status.CREATED)
-                .entity(processor.toWrapper(newDocument, linker.buildApplicationLinks(uriInfo, newDocument)))
-                .build();
+        try {
+            final OpenApiDocument document = repository.insert(swagger);
+            final OpenApiDocument newDocument = repository.findByApplicationIdWithEndpoints(document.getId());
+            return Response.status(Response.Status.CREATED)
+                    .entity(processor.toWrapper(newDocument, linker.buildApplicationLinks(uriInfo, newDocument)))
+                    .build();
+        } catch (TransactionalException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
we need a way to say why in the GUI "conflicting key", "optimistic locking", ... Not always possible but main cases will give value to the error instead of letting the user in the unknown probably. An ExceptionMapper is probably better since it would apply to all the app and not only there (these exceptions are shared by all CRUD endpoints)